packages/types/src/tx/schema/bond.ts

```diff
@@ -1,42 +1,28 @@
-import BN from "bn.js";
-import { Schema } from "borsh";
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import BigNumber from "bignumber.js";
+import { field } from "@dao-xyz/borsh";
+import { BigNumberSerializer } from "./utils";
+import { TxMsgValue } from "./tx";
 import { SubmitBondProps } from "../types";
-import { TxMsgSchema, TxMsgValue } from "./tx";
-import { SchemaObject } from "@anoma/utils";
 
 export class SubmitBondMsgValue {
-  source: string;
-  validator: string;
-  amount: BN;
-  native_token: string;
-  tx: TxMsgValue;
+  @field({ type: "string" })
+  source!: string;
 
-  constructor(properties: SubmitBondProps | SchemaObject<typeof BondMsgSchema>) {
-    this.source = properties.source;
-    this.validator = properties.validator;
-    this.amount = new BN(properties.amount.toString());
-    this.native_token = "nativeToken" in properties ?
-      properties.nativeToken :
-      properties.native_token;
-    this.tx = new TxMsgValue(properties.tx);
-  }
-}
+  @field({ type: "string" })
+  validator!: string;
+
+  @field(BigNumberSerializer)
+  amount!: BigNumber;
 
-export const BondMsgSchema = [
-  SubmitBondMsgValue,
-  {
-    kind: "struct",
-    fields: [
-      ["source", "string"],
-      ["validator", "string"],
-      ["amount", "u64"],
-      ["native_token", "string"],
-      ["tx", TxMsgValue],
-    ],
-  },
-] as const; // needed for SchemaObject to deduce types correctly
+  @field({ type: "string" })
+  nativeToken!: string;
 
-export const SubmitBondMsgSchema = new Map([
-  TxMsgSchema as [unknown, unknown],
-  BondMsgSchema as [unknown, unknown],
-]) as Schema;
+  @field({ type: TxMsgValue })
+  tx!: TxMsgValue;
+
+  constructor(data: SubmitBondProps) {
+    Object.assign(this, data);
+    this.tx = new TxMsgValue(data.tx);
```

Comment:
What does this line do? I thought `tx` would already be set by the `Object.assign` call above.

Reply:
I think we could actually use `TxMsgValue` to construct plain objects since TypeScript just sees classes as another type:

```
const props: TxMsgValue = {
  token: "token",
  feeAmount: new BigNumber(0),
  gasLimit: new BigNumber(0),
  chainId: "chainId",
};
```

This will typecheck just fine, but I don't know if that's intuitive behaviour :smile: